dhall/src/Dhall/Deriving.hs

```diff
@@ -0,0 +1,548 @@
+{-# LANGUAGE AllowAmbiguousTypes  #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE PolyKinds            #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-| Newtypes for writing customizable 'FromDhall' and 'ToDhall' instances
+    through the DerivingVia strategy.
+
+    Inspired by Matt Parson's blog post
+    [Mirror Mirror: Reflection and Encoding Via](https://www.parsonsmatt.org/2020/02/04/mirror_mirror.html),
+    but applied to Dhall instead of JSON.
+
+    This module is intended to be used with
+    [DerivingVia](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#extension-DerivingVia)
+    so it's only available for GHC >= v8.6.1.
+
+    Check "Dhall.Deriving#derivingVia" if you want to see this module in action.
```

Comment:
The way this anchor link is rendered is a bit confusing:

![image](https://user-images.githubusercontent.com/3664523/77028937-99891780-699a-11ea-8638-83d88182bb4c.png)

```suggestion
    Check the section /Letting DerivingVia do the work/ if you want to see this module in action.
```

Reply:
yeah, I wasn't sure about this one, it seems like a choice between having the right name but no link or the link but with the confusing name (why would you go to the same module you're already reading??). I'll try to reword it so we can have both.